util/webhook/webhook.go

```diff
@@ -173,25 +226,103 @@ func (a *ArgoCDWebhookHandler) HandleEvent(payload interface{}) {
 		}
 
 		for _, app := range apps.Items {
-			if !repoRegexp.MatchString(app.Spec.Source.RepoURL) {
-				log.Debugf("%s does not match", app.Spec.Source.RepoURL)
-				continue
-			}
-			targetRev := app.Spec.Source.TargetRevision
-			if targetRev == "HEAD" || targetRev == "" {
-				if !touchedHead {
-					continue
+			if appRevisionHasChanged(&app, revision, touchedHead) && appUsesURL(&app, webURL, repoRegexp) {
+				if appFilesHaveChanged(&app, changedFiles) {
+					_, err = argo.RefreshApp(appIf, app.ObjectMeta.Name, v1alpha1.RefreshTypeNormal)
+					if err != nil {
+						log.Warnf("Failed to refresh app '%s' for controller reprocessing: %v", app.ObjectMeta.Name, err)
+						continue
+					}
+				} else if change.shaBefore != "" && change.shaAfter != "" {
+					var cachedManifests cache.CachedManifestResponse
```

Comment:
What is the purpose of this block?
 ```else if change.shaBefore != "" && change.shaAfter != "" ```

Reply:
We cannot retrieve previous and next commit SHA for bitbucket, so these fields might be empty. This is to make sure we don't attempt to load/create cache for an empty string instead of SHA.